packages/ace-linters/message-controller.ts

```diff
@@ -12,27 +12,27 @@ import {
     ConfigureFeaturesMessage,
     ValidateMessage
 } from "./message-types";
-import * as oop from "ace-code/src/lib/oop";
-import {EventEmitter} from "ace-code/src/lib/event_emitter";
 import {IMessageController} from "./types/message-controller-interface";
 import * as lsp from "vscode-languageserver-protocol";
 import {CompletionService, ServiceFeatures, ServiceOptions, ServiceOptionsMap, SupportedServices} from "./types";
+import EventEmitter from "events";
 
-export class MessageController implements IMessageController {
+export class MessageController extends EventEmitter implements IMessageController {
     private $worker: Worker;
 
     constructor(worker: Worker) {
+        super();
         this.$worker = worker;
 
         this.$worker.addEventListener("message", (e) => {
             let message = e.data;
-            this["_signal"](message.type + "-" + message.sessionId, message.value);
+            this.emit(message.type + "-" + message.sessionId, message.value);
```

Comment:
can't we remove the event emitter use from here?

Reply:
we need it to direct message to specific editor session's callback

Reply:
Don't want to complicate it more than it's already, we could save this for other PR. Seems this change would be out of scope - making ace-code external